### DISCUSSION TITLE
Use variables in upload-artifact

### DISCUSSION THREAD

**Discussion by Subsurface-divelog** [community/community#25832] [Actions]
<p>This seems like it should be super simple:</p>
<pre><code>-uses:actions/upload-artifact@masterwith:
name:my-artifact-${GITHUB\_SHA}path:path/to/artifact
</code></pre>
<p>Except this doesn’t work - I get quite literally <code>my-artifact-${GITHUB_SHA}</code> as name, instead of a name that includes the SHA of the commit on which this is run… </p>

**Comment by dawidd6** [comment, marked as answer]
<p>Use:</p>
<pre><code>${{github.sha}}
</code></pre>

**Comment by Subsurface-divelog** [comment]
<p>I’ll try that.</p>
<p>Follow up question - is that documented anywhere?</p>

**Comment by dawidd6** [comment]
<p><a href="https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions" rel="nofollow noopener">https://help.github.com/en/articles/contexts-and-expression-syntax-for-github-actions</a></p>

**Comment by pmoura** [comment]
<p>It woulf be quite useful if we could access the runner context to write e.g. ${{ <span>runner.home }}</span>. This would avoid hard-coded paths when using the upload action:</p>
<pre><code>uses: actions/upload-artifact@master
        with:
          name: spider
          path: /Users/runner/collect/spider
</code></pre>
